dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/utils/SimpleReferenceCache.java

```diff
@@ -67,7 +67,7 @@
 
     private final Map<String, List<ReferenceConfigBase<?>>> referenceKeyMap = new ConcurrentHashMap<>();
     private final Map<Class<?>, List<ReferenceConfigBase<?>>> referenceTypeMap = new ConcurrentHashMap<>();
-    private final Map<ReferenceConfigBase<?>, Object> references = new ConcurrentHashMap<>();
+    private final Map<String, Object> references = new ConcurrentHashMap<>();
```

Comment:
It seems that the `type of key` does not need to be changed, the `equals` and `toString` methods are rewritten in `AbstractConfig`(`ReferenceConfigBase` is sub class of `AbstractConfig`)

Reply:
Change the title of pr, and simply declare the content of this change

Reply:
I think `type of key`  need to be changed, otherwise `KeyGenerator` makes no sense.

Reply:
@hanl754 
Agree with @BurningCN , the reason for not changing here is to consider multi-instance scenarios.